### ISSUE TITLE
How to load pre-trained model mamba_370m

### ISSUE THREAD

**Issue by Doctor-James** [hp-l33/AiM#8]
Thank you for your excellent work.
I would like to try training AiM from scratch. For example, when training AiM_L, I encountered many size and key mismatches while loading the pre-trained model from mamba_370m. Could you provide some suggestions on how to resolve this? How did you handle the model loading?

**Comment by Doctor-James** [comment]
It might be related to the version of mamba_ssm. Could you share which version of mamba_ssm you're using and which pre-trained model you loaded? For reference, I'm using [mamba-370m](https://huggingface.co/state-spaces/mamba-370m) and mamba_ssm==2.2.2

**Comment by hp-l33** [comment]
Hi! Unfortunately, due to several modifications we made to the model architecture—such as incorporating class embeddings and AdaLN—the weights of AiM do not precisely match those of the original mamba model. In fact, when training AiM, we started entirely from scratch and did not load any pre-trained weights.

**Comment by hp-l33** [comment]
The configuration of training script parameters is crucial. Here are the specific parameters we used for training AiM-L, for your reference:
```
accelerate launch --num_processes=16 --num_machines=2 --main_process_ip=... --main_process_port=... --machine_rank=... train_stage2.py --aim-model AiM-L --dataset /your/data/path/ --vq-ckpt /your/ckpt/path/vq_f16.pt --batch-size 128 --lr 8e-4 --epochs 300
```
Please note that if the global batch size changes, the learning rate should be adjusted accordingly. We followed a setting of 1e-4 per 256 batch size. In the script above, we used 16 GPUs, with each GPU having a batch size of 128, making the total batch size 2048. Therefore, the learning rate is calculated as 2048 / 256 * 1e-4 = 8e-4.

**Comment by Doctor-James** [comment]
Thank you for your suggestion. I made a mistake earlier and realized that you're using the mamba2 model. After switching to mamba2_370m, I was able to load most of the weights. The main missing weights are for adaln and position_embeddings. I believe partially loading the weights might still be effective. Do you have any recommendations?

**Comment by hp-l33** [comment]
I'm glad to see that you've resolved the issue. I'm also curious whether loading the pre-trained weights trained on language datasets would be beneficial. Looking forward to your experiments, and best wishes!

**Comment by Doctor-James** [comment]
May I ask about the loss curve throughout your whole training process? After approximately how many epochs or when the loss reaches a certain point, can a relatively normal image be generated? I’m training AiM-B with a batch size of 1024, learning rate of 4e-4, and at 80k steps, the loss is around 6.3, but the quality of the generated images is still very poor, barely recognizable.

Additionally, I found that loading the pre-trained Mamba2-130M model doesn't perform as well as training from scratch.

**Comment by hp-l33** [comment]
That sounds strange. In our experiments, even with the largest model, the loss never dropped below 6.5. The final loss of the AiM-B we trained was 7.399, and it could generate recognizable images after around 50 epochs (with the loss slightly below 7.6). I’ve shared the training loss of AiM-B on wandb for your reference.
https://api.wandb.ai/links/1370922505/tk5by16n
